client/dom/ColorScale/ColorScale.ts

```diff
@@ -31,22 +31,23 @@ export class ColorScale {
 	tickSize: number
 	tickValues: number[]
 	topTicks: boolean
+	usePercentiles: boolean
 
 	constructor(opts: ColorScaleOpts) {
 		this.barheight = opts.barheight || 14
 		this.barwidth = opts.barwidth || 100
 		this.colors = opts?.colors?.length ? opts.colors : ['white', 'red']
-		this.data = opts.data
+		this.domain = opts.domain
 		this.fontSize = opts.fontSize || 10
 		this.markedValue = opts.markedValue && opts.markedValue > 0.001 ? opts.markedValue : null
 		this.ticks = opts.ticks || 5
 		this.tickSize = opts.tickSize || 1
 		this.topTicks = opts.topTicks || false
-		this.domain = opts.domain
+		this.usePercentiles = opts.usePercentiles || false
 
 		this.validateOpts(opts)
-
-		this.tickValues = niceNumLabels(opts.data)
+		if (opts.usePercentiles) opts.domain = removeExtremeOutliers(opts.domain)
```

Comment:
i'm thinking calculating cutoff value by a percentile should be carried out outside of ColorScale. 

also in cnv legend, caller supplies opts.domain[] array with the axis start/stop, but not the actual long array of cnv values for all cnv events, thus should not do percentile calculation here